src/System.Drawing.Common/src/System/Drawing/GdiPlusStreamHelper.Unix.cs

```diff
@@ -41,164 +41,80 @@ namespace System.Drawing
 {
     internal sealed partial class GdiPlusStreamHelper
     {
-        public Stream stream;
+        private Stream _stream;
 
-        private StreamGetHeaderDelegate sghd = null;
-        private StreamGetBytesDelegate sgbd = null;
-        private StreamSeekDelegate skd = null;
-        private StreamPutBytesDelegate spbd = null;
-        private StreamCloseDelegate scd = null;
-        private StreamSizeDelegate ssd = null;
-        private byte[] start_buf;
-        private int start_buf_pos;
-        private int start_buf_len;
-        private byte[] managedBuf;
-        private const int default_bufsize = 4096;
-
-        public GdiPlusStreamHelper(Stream s, bool seekToOrigin)
-        {
-            managedBuf = new byte[default_bufsize];
-
-            stream = s;
-            if (stream != null && stream.CanSeek && seekToOrigin)
-            {
-                stream.Seek(0, SeekOrigin.Begin);
-            }
-        }
-
-        public int StreamGetHeaderImpl(IntPtr buf, int bufsz)
+        public unsafe GdiPlusStreamHelper(Stream stream, bool seekToOrigin)
         {
-            int bytesRead;
-
-            start_buf = new byte[bufsz];
-
-            try
+            // Seeking required
+            if (!stream.CanSeek)
             {
-                bytesRead = stream.Read(start_buf, 0, bufsz);
+                var memoryStream = new MemoryStream();
+                stream.CopyTo(memoryStream);
+                _stream = memoryStream;
             }
-            catch (IOException)
+            else
             {
-                return -1;
+                _stream = stream;
             }
 
-            if (bytesRead > 0 && buf != IntPtr.Zero)
+            if (seekToOrigin)
             {
-                Marshal.Copy(start_buf, 0, (IntPtr)(buf.ToInt64()), bytesRead);
+                _stream.Seek(0, SeekOrigin.Begin);
             }
 
-            start_buf_pos = 0;
-            start_buf_len = bytesRead;
-
-            return bytesRead;
+            CloseDelegate = new StreamCloseDelegate(StreamCloseImpl);
```

Comment:
nit: shouldn't need to new up, should be implicit I believe (e.g. `CloseDelegate = StreamCloseImpl`)